File: backend/app/models/retention.py
"""Pydantic models for retention policy API."""

from typing import Optional

from pydantic import BaseModel, Field


class RetentionPolicyCreate(BaseModel):
    category: str = Field(..., description="Data category (e.g. execution_logs)")
    scope: str = Field("global", description="Scope: global, team, or bot")
    scope_name: str = Field("All Teams", description="Human-readable scope name")
    retention_days: int = Field(90, ge=1, description="Number of days to retain data")
    delete_on_expiry: bool = Field(True, description="Delete data on expiry")
    archive_on_expiry: bool = Field(False, description="Archive data on expiry")
    estimated_size_gb: float = Field(0.0, ge=0.0, description="Estimated data size in GB")


class RetentionPolicyToggle(BaseModel):
    enabled: bool = Field(..., description="Enable or disable the policy")


class RetentionPolicyResponse(BaseModel):
    id: str
    category: str
    scope: str
    scope_name: str
    retention_days: int
    delete_on_expiry: int
    archive_on_expiry: int
    estimated_size_gb: float
    enabled: int
    created_at: Optional[str] = None


class PolicyIdPath(BaseModel):
    policy_id: str = Field(..., description="Retention policy ID")

File: frontend/src/services/api/retention.ts
/**
 * Retention policy API module.
 */
import { apiFetch } from './client';

export interface RetentionPolicy {
  id: string;
  category: string;
  scope: string;
  scope_name: string;
  retention_days: number;
  delete_on_expiry: number;
  archive_on_expiry: number;
  estimated_size_gb: number;
  enabled: number;
  created_at?: string;
}

export interface CreateRetentionPolicyRequest {
  category: string;
  scope?: string;
  scope_name?: string;
  retention_days?: number;
  delete_on_expiry?: boolean;
  archive_on_expiry?: boolean;
  estimated_size_gb?: number;
}

export const retentionApi = {
  list: (): Promise<{ policies: RetentionPolicy[] }> =>
    apiFetch<{ policies: RetentionPolicy[] }>('/admin/retention-policies/'),

  create: (data: CreateRetentionPolicyRequest): Promise<RetentionPolicy> =>
    apiFetch<RetentionPolicy>('/admin/retention-policies/', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  toggle: (policyId: string, enabled: boolean): Promise<{ id: string; enabled: boolean }> =>
    apiFetch<{ id: string; enabled: boolean }>(
      `/admin/retention-policies/${policyId}/toggle`,
      {
        method: 'PATCH',
        body: JSON.stringify({ enabled }),
      }
    ),

  delete: (policyId: string): Promise<void> =>
    apiFetch<void>(`/admin/retention-policies/${policyId}`, {
      method: 'DELETE',
    }),

  runCleanup: (): Promise<{ message: string }> =>
    apiFetch<{ message: string }>('/admin/retention-policies/cleanup', {
      method: 'POST',
    }),
};
